src/components/PopupWithForm.js

```diff
@@ -21,12 +36,33 @@ export default class PopupWithForm extends Popup {
     this._form.addEventListener("submit", (event) => {
       event.preventDefault();
       this._handleFormSubmit(this._getInputValues());
-      this.close();
+      const result = this._handleFormSubmit(this._getInputValues());
+      if (result instanceof Promise) {
+        result
+          .then(() => {
+            this.close();
+          })
+          .catch((error) => {
+            console.error("Error during confirmation:", error);
+          });
+      } else {
+        this.close();
+      }
     });
   }
+  //
+
+  _isValidForm() {
+    const inputValues = this._getInputValues();
+    if (!inputValues.avatar) {
```

Comment:
Please, delete `_isValidForm` because you check `avatar` here. No all forms have avatar inputs

There should **not** be any code in universal functions that doesn’t suit every popup.